### ISSUE TITLE
Test integration with current Cylc GraphQL endpoint

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-web#43]
* Description: https://github.com/cylc/cylc/issues/2900#issuecomment-468663845
* Code: https://github.com/dwsutherland/cylc/tree/flask-gevent-graphql

We should be able to use the current Apollo Client to connect to the API. Things to consider are CORS, headers, authentication (how/where/when it would be done), interceptors (see #40 ), error handling, and also test the one mutation provided to stop suites.

The port of the Cylc process also needs to be unique, so we will need to limit the range to a single port number.

**Comment by kinow** [comment]
Ran into issues trying to run suite `five`. Reported in https://github.com/cylc/cylc/issues/2900, and will try it again later this week.

**Comment by kinow** [comment]
For now added the code to run `cylc scan` and display the suites (thanks to @dwsutherland for the original code).

![image](https://user-images.githubusercontent.com/304786/53850826-730f6d00-4021-11e9-93ce-1e9a4f533caa.png)

Next step is to update the existing GraphQL code to query the backend.

**Comment by kinow** [comment]
Now we can add an icon or link to the table, linking to something like `/users/kinow/suites/five`. That view will call the same `SuiteService`, but a different method.

And that method will execute a GraphQL, like this one:

```
query {
  tasks {
    id, name, meanElapsedTime, namespace, depth
  }
}
```

Sample response:

```
{
  "data": {
    "tasks": [
      {
        "id": "foo",
        "name": "foo",
        "meanElapsedTime": null,
        "namespace": [
          "root",
          "foo"
        ],
        "depth": 1
      },
      {
        "id": "bar",
        "name": "bar",
        "meanElapsedTime": null,
        "namespace": [
          "root",
          "bar"
        ],
        "depth": 1
      },
      {
        "id": "prep",
        "name": "prep",
        "meanElapsedTime": null,
        "namespace": [
          "root",
          "prep"
        ],
        "depth": 1
      }
    ]
  }
}
```

**Comment by kinow** [comment]
Success! Mainly thanks to the patience of @dwsutherland for taking so many of my requests and promptly fixing or pointing what I had to do in the Vue.js app (big thanks).

Now in /suites we have a normal HTTP REST query, to a local Tornado handler that executes `cylc scan`. The results are then populated in a table.

If the user chooses one of the running suites (note: later we need to add states, and test with multiple suites), then we display /suites/$suite-name.

In this last screen, the user will have a table populated via the GraphQL endpoint. In this case, we are defaulting to `http://${suite-host}:{suite-port}/graphql`, with the query

```
query {
  tasks {
    id, name, meanElapsedTime, namespace, depth
  }
}
```

I've put my suite on hold to test. Releasing the suite won't update the screen, as we are not using websockets and/or graphql subscriptions. Pagination is also done client side, but it could be improved later and use graphql pagination to save server resources.

But for now, I must say I am impressed by how easy it was to use @dwsutherland 's API. Great work!

Coverage increased a bit as I've added more tests to Vuex store modules. Closing now.
